org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/TolerantJavaCompiler.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+* Copyright (c) 2024 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.internal.javac;
+
+import java.util.Queue;
+
+import javax.tools.JavaFileObject;
+
+import com.sun.tools.javac.comp.AttrContext;
+import com.sun.tools.javac.comp.CompileStates.CompileState;
+import com.sun.tools.javac.comp.Env;
+import com.sun.tools.javac.main.JavaCompiler;
+import com.sun.tools.javac.tree.JCTree.JCClassDecl;
+import com.sun.tools.javac.util.Context;
+import com.sun.tools.javac.util.Pair;
+
+public class TolerantJavaCompiler extends JavaCompiler {
+	boolean isInGeneration = false;
+
+	public TolerantJavaCompiler(Context context) {
+		super(context);
+	}
+
+	@Override
+	protected boolean shouldStop(CompileState cs) {
+		// Never stop
+		return false;
+	}
```

Comment:
It looks like we could replace it with `options.put("should-stop.ifError", CompileState.GENERATE)` earlier when configuring context.

Reply:
this class is reused by the build job. In the build job, we don't want to stop generate if error.